### PR TITLE
feat(rate-limit): remove group key support and improve global rate limit

### PR DESCRIPTION
**Pull request by manast**
BREAKING CHANGE: limit by group keys has been removed in favor of a much simpler and efficent rate-limit implementation.

### CODE REVIEW

File: tests/test_flow.ts [hunk 1]
```diff
@@ -1266,7 +1266,7 @@ describe('flows', () => {
     });
   });
 
-  it('should rate limit by grouping', async function () {
+  it.skip('should rate limit by grouping', async function () {
```
Comment (roggervalf) [inline tests/test_flow.ts:1269]: could we remove this test?

File: tests/test_rate_limiter.ts [hunk 2]
```diff
@@ -154,6 +154,66 @@ describe('Rate Limiter', function () {
...
+
+    const numJobs = 10;
+    const dynamicLimit = 250;
+    const duration = 100;
+
+    const worker = new Worker(
+      queueName,
+      async job => {
+        if (job.attemptsMade === 1) {
+          await worker.rateLimit(dynamicLimit);
```
Comment (roggervalf) [inline tests/test_rate_limiter.ts:168]: could we do: await this.rateLimit(dynamicLimit);

**Review by roggervalf — APPROVED** [review]
LGTM

File: tests/test_rate_limiter.ts (hunk 2 above)
Comment (manast) [reply tests/test_rate_limiter.ts:168]: No, it won't work with this. But it is a good point, we need to support "this" in order to support sandboxed processors.

Comment (JCMais) [reply tests/test_rate_limiter.ts:168]: just reading this, so, ratelimit is not supported for sandboxed processors?

**Pull request merged**